tests/functional/sse-kms-migration/arnPrefix.js

```
@@ -505,3 +506,163 @@ describe('ensure MPU use good SSE', () => {
+    };
+    // localkms container returns a different error message when the key is pending deletion
+    // as we decrypt without passing the keyId, so we need to handle it separately
+    const localKms = 'The ciphertext refers to a customer master key that does not exist, ' +
+        'does not exist in this region, or you are not allowed to access\\.';
+    const expectedAWS = {
+        code: /KMS\.KMSInvalidStateException|KMS\.AccessDeniedException/,
+        msg: (_, keyId) => new RegExp(`${keyId} is pending deletion\\.|${localKms}`),
+    };
+
+    if (helpers.config.backends.kms === 'kmip') {
+        ({ code: expectedErr, msg: expectedMsg } = expectedKMIP);
+    } else if (helpers.config.backends.kms === 'aws') {
+        ({ code: expectedErr, msg: expectedMsg } = expectedAWS);
+    } else {
+        throw new Error(`Unsupported KMS backend: ${helpers.config.backends.kms}`);
+    }
+
+    function assertKmsError(action, keyId) {
+        return err => {
+            assert.match(err.name, expectedErr);
+            assert.match(err.message, expectedMsg(action, keyId));
+            return true;
+        };
+    }
+
+    before(async () => {
+        void await helpers.s3.createBucket({ Bucket }).promise();
+        await helpers.s3.putObject({
+            ...helpers.putObjParams(Bucket, 'plaintext', {}, null),
+            Body: body,
+        }).promise();
+
+        ({ masterKeyId, masterKeyArn } = await helpers.createKmsKey(log));
+        await helpers.putEncryptedObject(Bucket, Key, sseConfig, masterKeyArn, body);
+        // ensure we can decrypt and read the object
+        const obj = await helpers.s3.getObject({ Bucket, Key }).promise();
+        assert.strictEqual(obj.Body.toString(), body);
+        void await destroyKmsKey(masterKeyArn, log);
+    });
+
+    after(async () => {
+        void await helpers.cleanup(Bucket);
+        if (masterKeyArn) {
+            try {
+                void await destroyKmsKey(masterKeyArn, log);
```

Dropping the awaited value with `void` swallows any rejection; await directly (or wrap in try/catch + log) so cleanup failures surface in CI.

---

void only replaces the expression returned value with `undefined`. It does not swallow any `await`ed rejection throws. Here the `try catch` does actually catch and ignore an error.

I added a void for eslint rule https://eslint.org/docs/latest/rules/no-unused-expressions

---

I'll make another PR to replace this with ignoring specific linting rule

---

There https://github.com/scality/cloudserver/pull/5847